File: MathEditor/mathEditor/internal/MTView/MTView+AutoLayout.m
//
//  MTView+AutoLayout.m
//  MathEditor
//
//  Created by Madiyar Aitbayev on 20/03/2026.
//

#import "MTConfig.h"
#import "MTView+AutoLayout.h"

@implementation MXView (AutoLayout)

- (void)pinToSuperview
{
    [self pinToSuperviewWithTop:0 leading:0 bottom:0 trailing:0];
}

- (void)pinToSuperviewWithTop:(CGFloat)top leading:(CGFloat)leading bottom:(CGFloat)bottom trailing:(CGFloat)trailing
{
    MTView *superview = self.superview;
    if (!superview)
        return;
    self.translatesAutoresizingMaskIntoConstraints = NO;
    [NSLayoutConstraint activateConstraints:@[
        [self.topAnchor constraintEqualToAnchor:superview.topAnchor constant:top],
        [self.leadingAnchor constraintEqualToAnchor:superview.leadingAnchor constant:leading],
        [self.trailingAnchor constraintEqualToAnchor:superview.trailingAnchor constant:-trailing],
        [self.bottomAnchor constraintEqualToAnchor:superview.bottomAnchor constant:-bottom]
    ]];
}

@end

File: mathEditorSwift/Internal/MTView/MTView+AutoLayout.swift
//
//  MTView+AutoLayout.swift
//  MathEditor
//
//  Created by Madiyar Aitbayev on 24/03/2026.
//

import Foundation

#if canImport(AppKit)
  import AppKit
#elseif canImport(UIKit)
  import UIKit
#endif

extension MTView {
  @objc public func pinToSuperview() {
    pinToSuperview(withTop: 0, leading: 0, bottom: 0, trailing: 0)
  }

  @objc public func pinToSuperview(
    withTop top: CGFloat, leading: CGFloat, bottom: CGFloat, trailing: CGFloat
  ) {
    guard let superview else { return }
    translatesAutoresizingMaskIntoConstraints = false

    NSLayoutConstraint.activate([
      topAnchor.constraint(equalTo: superview.topAnchor, constant: top),
      leadingAnchor.constraint(equalTo: superview.leadingAnchor, constant: leading),
      trailingAnchor.constraint(equalTo: superview.trailingAnchor, constant: -trailing),
      bottomAnchor.constraint(equalTo: superview.bottomAnchor, constant: -bottom),
    ])
  }
}
